docs/guides/materialize_source_table.md

```diff
@@ -0,0 +1,195 @@
+
+## Manual Migration Process
+
+This manual migration process requires users to create new source tables using the new `materialize_source_table_{source}` resource and then remove the old ones. In this example, we will use MySQL as the source type.
+
+### Step 1: Define `materialize_source_table_mysql` Resources
+
+Before making any changes to your existing source resources, create new `materialize_source_table_mysql` resources for each table that is currently defined within your sources. This ensures that the tables are preserved during the migration:
+
+```hcl
+resource "materialize_source_table_mysql" "mysql_table_from_source" {
+  name           = "mysql_table1_from_source"
+  schema_name    = "public"
+  database_name  = "materialize"
+
+  source {
+    name = materialize_source_mysql.mysql_source.name
+    // Define the schema and database for the source if needed
+  }
+
+  upstream_name        = "mysql_table1"
+  upstream_schema_name = "shop"
+
+  ignore_columns = ["about"]
+}
+```
+
+### Step 2: Apply the Changes
+
+Run `terraform plan` and `terraform apply` to create the new `materialize_source_table_mysql` resources. This step ensures that the tables are defined separately from the source and are not removed from Materialize.
+
+> **Note:** This will start an ingestion process for the newly created source tables.
+
+### Step 3: Remove Table Blocks from Source Resources
+
+Once the new `materialize_source_table_mysql` resources are successfully created, you can safely remove the `table` blocks from your existing source resources:
+
+```hcl
+resource "materialize_source_mysql" "mysql_source" {
+  name         = "mysql_source"
+  cluster_name = "cluster_name"
+
+  mysql_connection {
+    name = materialize_connection_mysql.mysql_connection.name
+  }
+
+  // Remove the table blocks from here
+  - table {
+  -   upstream_name        = "mysql_table1"
+  -   upstream_schema_name = "shop"
+  -   name                 = "mysql_table1_local"
+  -
+  -   ignore_columns = ["about"]
+  -
+  ...
+}
+```
+
+This will drop the old tables from the source resources.
+
+### Step 4: Update Terraform State
+
+After removing the `table` blocks from your source resources, run `terraform plan` and `terraform apply` again to update the Terraform state and apply the changes.
+
+### Step 5: Verify the Migration
+
+After applying the changes, verify that your tables are still correctly set up in Materialize by checking the table definitions using Materialize’s SQL commands.
+
+During the migration, you can use both the old `table` blocks and the new `materialize_source_table_{source}` resources simultaneously. This allows for a gradual transition until the old method is fully deprecated.
+
+The same approach can be used for other source types such as Postgres, eg. `materialize_source_table_postgres`.
+
+## Automated Migration Process (TBD)
```

Comment:
nice - this is great! We will probably want to figure out how to tell them that they will be able to coordinate the 'automated' migration process with their field-engineer representative if they go down this path